### PR TITLE
Add utils gitlab 1 to implement 1

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no way to fetch gitlab data via web scraping. #34 
- Currently there is no way to process gitlab data. #35

**What is the new behavior (if this is a feature change)?**
- Now there is a way to fetch gitlab data via web scraping. #34 
- Now there is a way to process gitlab data using the OSM models. #35

**Other information:**
Ref: https://gitlab.com
Ref: 🐍

### CODE REVIEW

File: src/utils/gitlab/index.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import * as converter from './converter';
+//#endregion
+
+//#region > Exports
+export { paths, converter };
+//#endregion
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © Simon Prast
```
Comment (pinterid) [inline src/utils/gitlab/index.ts:16]: Update License

File: src/utils/gitlab/paths.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const projects = (username: string) => home(username) + "/projects.json";
+//#endregion
+
+//#region > Exports
+export { home, currentCalendar, activity, atom, groups, projects };
+//#endregion
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © Simon Prast
```
Comment (pinterid) [inline src/utils/gitlab/paths.ts:18]: Update License.

File: src/utils/gitlab/converter.ts [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+}
+//#endregion
+
+//#region > Exports
+export { runScraper };
+//#endregion
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © Simon Prast
```
Comment (pinterid) [inline src/utils/gitlab/converter.ts:407]: Update license.

File: src/utils/gitlab/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//#region > Imports
+//> Paths
+// Contains all path variables
+import * as paths from "./paths";
+//> Convertor
```
Comment (kleberbaum) [inline src/utils/gitlab/index.ts:5]: Converter? Please look for other cases of this typo.

File: src/utils/gitlab/converter.ts [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+  let eventList = events.getElementsByClassName("event-item");
+  let days: CalendarDays = {};
+  let stats: Statistic = {};
+
+  /* Check whether there are contributions/events or not */
+  if (eventList) {
+    /*
+      Loop trough every item in the event list.
+      Extract contribution data from element.
+    */
```
Comment (kleberbaum) [inline src/utils/gitlab/converter.ts:215]: Should be more like:
```
    /* 
     * Set the organization as owner for each repository within
     * the organization. To render a organization as a member
     * a list of member keys to render by is defined.
     */
```

File: src/utils/gitlab/converter.ts [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+        .getElementsByClassName("event-scope")[0]
+        ?.getElementsByTagName("a")[0]
+        .getAttribute("href");
+
+      if (!name) {
+        name = element
+          .getElementsByClassName("event-title")[0]
+          .getElementsByTagName("a")[0]
+          .getAttribute("href");
+        if (!name) {
```
[the remark this answers is not in the document]
Comment (schettn) [reply src/utils/gitlab/converter.ts:244]: Cannot rewrite it to an inline if because ts sucks.

**Pull request merged**